src/rust/src/metrics_ffi.rs

```diff
@@ -0,0 +1,185 @@
+use libc::{c_char, c_double};
+use metrics::{try_recorder, GaugeValue, Key, KeyData, Label};
+use metrics_exporter_prometheus::PrometheusBuilder;
+use std::ffi::CStr;
+use std::net::SocketAddr;
+use std::ptr;
+use std::slice;
+use tracing::error;
+
+#[no_mangle]
+pub extern "C" fn metrics_run(listen_address: *const c_char) -> bool {
+    let listen_address = unsafe { CStr::from_ptr(listen_address) }.to_str().unwrap();
```

Comment:
I think this is a reachable panic if `listen_address` (taken from a command option) is not valid UTF-8.